src/Common/Admin/Help_Hub/Hub.php

```diff
@@ -0,0 +1,537 @@
+	 *
+	 * @since TBD
+	 *
+	 * @var Help_Hub_Data_Interface
+	 */
+	protected Help_Hub_Data_Interface $data;
+
+	/**
+	 * The template class.
+	 *
+	 * @since TBD
+	 *
+	 * @var Tribe__Template
+	 */
+	protected Tribe__Template $template;
+
+	/**
+	 * Initializes configuration and necessary constants for the Help Hub.
+	 *
+	 * @since TBD
+	 */
+	public function __construct() {
+		$this->config   = tribe( Configuration::class );
+		$this->template = new Tribe__Template();
+	}
+
+	/**
+	 * Sets the data object for the Help Hub and registers hooks.
+	 *
+	 * Assigns a data object implementing Help_Hub_Data_Interface
+	 * to the $data property and registers necessary actions and filters.
+	 *
+	 * @since TBD
+	 *
+	 * @param Help_Hub_Data_Interface $data The data class instance containing Help Hub resources.
+	 *
+	 * @return self
+	 */
+	public function setup( Help_Hub_Data_Interface $data ): self {
+		$this->data = $data;
+		$this->register_hooks();
+
+		return $this;
+	}
+
+	/**
+	 * Retrieves the data object used by the Help Hub.
+	 *
+	 * @since TBD
+	 *
+	 * @return Help_Hub_Data_Interface The data object containing Help Hub resources.
+	 */
+	public function get_data(): Help_Hub_Data_Interface {
+		return $this->data;
+	}
+
+	/**
+	 * Registers the hooks and filters needed for Help Hub functionality.
+	 *
+	 * Sets up actions and filters for initializing iframe content,
+	 * loading assets, and adding custom body classes for Help Hub pages.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function register_hooks(): void {
+		add_action( 'admin_init', [ $this, 'generate_iframe_content' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'load_assets' ], 1 );
+		add_filter( 'admin_body_class', [ $this, 'add_help_page_body_class' ] );
+	}
+
+	/**
+	 * Ensures that the Help Hub data object is set.
+	 *
+	 * Verifies that the $data property has been set. Throws a RuntimeException
+	 * if the data has not been set using the setup method.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 * @throws RuntimeException If data has not been set using setup.
+	 */
+	protected function ensure_data_is_set(): void {
+		if ( empty( $this->data ) ) {
+			throw new RuntimeException( 'The HelpHub data must be set using the setup method before calling this function.' );
+		}
+	}
+
+	/**
+	 * Renders the Help Hub page.
+	 *
+	 * Generates necessary notices, retrieves data, and renders the appropriate Help Hub template.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 * @throws RuntimeException If data is not set using the setup method before rendering.
+	 */
+	public function render(): void {
+		/**
+		 * Fires before the Help Hub page is rendered.
+		 *
+		 * Use this hook to modify data or enqueue additional assets before the Help Hub template is generated.
+		 *
+		 * @since TBD
+		 *
+		 * @param Hub $this The Hub instance.
+		 */
+		do_action( 'tec_help_hub_before_render', $this );
+
+		$this->ensure_data_is_set();
```

Comment:
Feel like this should be before the above hook - since it will throw an exception if it fails.